index.js

```diff
@@ -227,22 +227,16 @@ Config.default = function() {
 };
 
 Config.detect = function(options, filename) {
-  // Only attempt to detect the backup if a specific file wasn't asked for.
-  var checkBackup = false;
+  var search;
 
-  if (filename == null) {
-    filename = DEFAULT_CONFIG_FILENAME;
-    checkBackup = true;
-  }
+  (filename == null)
```

Comment:
While we're at it, let's just check for filename not being falsey? This might be an existing bug cause omitted arguments are `undefined`?